jlm/llvm/ir/operators/call.hpp

```diff
@@ -219,7 +218,7 @@ class CallTypeClassifier final
   }
 
   static std::unique_ptr<CallTypeClassifier>
-  CreateNonRecursiveDirectCallClassifier(lambda::output & output)
+  CreateNonRecursiveDirectCallClassifier(rvsdg::output & output)
   {
     return std::make_unique<CallTypeClassifier>(CallType::NonRecursiveDirectCall, output);
```

Comment:
Added assert as well as documentation for pre-condition. I see that the only call site is actually "ClassifyCall", and that does exactly the same checks -- I guess compiler will be able to elide them, but since these convenience helpers are trivial might make sense to dissolve them into the caller.